### PR TITLE
Add metrics to weather_mv ee.

### PR DESCRIPTION
**Pull request by aniketsinghrawat**
This PR adds support to get the element_processing_time metrics to beam.

We are actually calculating how much time it takes for an element to be processed during each stage, this information is passed with the element in the PCollection itself.

This is done in three parts:

`AddTimer` DoFn: Add this DoFn above the stage you want to calculate the metrics from.
`timeit` decorator: _After_ the AddTimer stage, add this decorator to the `process` functions for all subsequent stages.
`AddMetrics` DoFn: Finally after the last stage, add this DoFn that adds the actual metrics.

### CODE REVIEW

File: weather_mv/loader_pipeline/metrics.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.metrics import metric
+
+
+def timeit(func_name: str, keyed_fn: bool = False):
+    """Decorator to add time it takes for an element to be processed by a stage.
+
+    Args:
+        func_name: A unique name of the stage.
+        keyed_fn (optional): This has to be passed true if the input is adding keys to the element.
```
Comment (deepgabani8) [inline weather_mv/loader_pipeline/metrics.py:16]: Can you add more info about this flag please? Like when it would be used.

**Review by deepgabani8 — APPROVED** [review]

File: weather_mv/loader_pipeline/metrics.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import time
```
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/metrics.py:1]: Please add license comment.

File: weather_mv/loader_pipeline/metrics.py (hunk 1 above)
Comment (aniketsinghrawat) [reply weather_mv/loader_pipeline/metrics.py:32]: done

File: weather_mv/loader_pipeline/metrics.py (hunk 2 above)
Comment (aniketsinghrawat) [reply weather_mv/loader_pipeline/metrics.py:17]: done

File: weather_mv/loader_pipeline/metrics.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    else:
+                        yield result, new_time_dict
+            else:
+                raise ValueError("Function is not a generator.")
+
+        return wrapper
+    return decorator
+
+
+class AddTimer(beam.DoFn):
```
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/metrics.py:81]: Add docstring.

File: weather_mv/loader_pipeline/util.py [hunk 4]
```diff
@@ -324,6 +325,15 @@ def _shard(elem, num_shards: int):
     return (np.random.randint(0, num_shards), elem)
 
 
+class Shard(beam.DoFn):
```
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/util.py:328]: nit: add docstring.

**Review by mahrsee1997 — APPROVED** [review]
LGTM. Few minor nits ...

File: weather_mv/loader_pipeline/ee.py
Comment (mahrsee1997) [inline weather_mv/loader_pipeline/ee.py:1]: Also do version bump in setup.py.

Comment (aniketsinghrawat) [reply weather_mv/loader_pipeline/ee.py:1]: done

File: weather_mv/loader_pipeline/util.py (hunk 4 above)
Comment (aniketsinghrawat) [reply weather_mv/loader_pipeline/util.py:328]: done

File: weather_mv/loader_pipeline/metrics.py (hunk 3 above)
Comment (aniketsinghrawat) [reply weather_mv/loader_pipeline/metrics.py:81]: done

**Pull request merged**